seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sqlengine/zeta/ZetaSimpleSQLEngine.java

```diff
@@ -0,0 +1,568 @@
+                if (leftVal == null || rightVal == null) {
+                    return false;
+                }
+                if (leftVal instanceof Number && rightVal instanceof Number) {
+                    return ((Number) leftVal).doubleValue() <= ((Number) rightVal).doubleValue();
+                }
+                if (leftVal instanceof String && rightVal instanceof String) {
+                    return ((String) leftVal).compareTo((String) rightVal) <= 0;
+                }
+                if (leftVal instanceof LocalDateTime && rightVal instanceof LocalDateTime) {
+                    return ((LocalDateTime) leftVal).isBefore((LocalDateTime) rightVal)
+                            || ((LocalDateTime) leftVal).isEqual((LocalDateTime) rightVal);
+                }
+                if (leftVal instanceof LocalDate && rightVal instanceof LocalDate) {
+                    return ((LocalDate) leftVal).isBefore((LocalDate) rightVal)
+                            || ((LocalDate) leftVal).isEqual((LocalDate) rightVal);
+                }
+                if (leftVal instanceof LocalTime && rightVal instanceof LocalTime) {
+                    return ((LocalTime) leftVal).isBefore((LocalTime) rightVal)
+                            || leftVal.equals(rightVal);
+                }
+                throw new TransformException(
+                        CommonErrorCode.UNSUPPORTED_OPERATION,
+                        String.format(
+                                "Filed types not matched, left: %s, right:%s ",
+                                leftVal.getClass().getSimpleName(),
+                                rightVal.getClass().getSimpleName()));
+            }
+        }
+        if (whereExpr instanceof AndExpression) {
+            AndExpression andExpression = (AndExpression) whereExpr;
+            Expression leftExpr = andExpression.getLeftExpression();
+            boolean leftRes = physicalFilter(leftExpr, inputFields);
+            Expression rightExpr = andExpression.getRightExpression();
+            boolean rightRes = physicalFilter(rightExpr, inputFields);
+            return leftRes && rightRes;
+        }
+        if (whereExpr instanceof OrExpression) {
+            OrExpression andExpression = (OrExpression) whereExpr;
+            Expression leftExpr = andExpression.getLeftExpression();
+            boolean leftRes = physicalFilter(leftExpr, inputFields);
+            Expression rightExpr = andExpression.getRightExpression();
+            boolean rightRes = physicalFilter(rightExpr, inputFields);
+            return leftRes || rightRes;
+        }
+        if (whereExpr instanceof Parenthesis) {
+            Parenthesis parenthesis = (Parenthesis) whereExpr;
+            Expression expression = parenthesis.getExpression();
+            return physicalFilter(expression, inputFields);
+        }
+        return false;
+    }
+
+    private Object[] physicalScanTable(SeaTunnelRow inputRow) {
```

Comment:
Seem like this logic shouldn't run on each row.

Reply:
Yes，you are right